emuu-server/video/video.go

```diff
@@ -131,20 +155,43 @@ func SetVideos(c *gin.Context) {
 			var vid = Video{}
 
 			doc.DataTo(&vid)
-			fmt.Printf("Document data: %#v", vid)
-			fmt.Println(doc.Data())
 			mostViewedArr = append(mostViewedArr, vid)
 			recentArr = append(recentArr, vid)
 		}
 		sortMostViewed(mostViewedArr)
 		sortRecent(recentArr)
+		fmt.Println(recentArr)
+
+		var pageAmount int
+		pageAmount = int(math.Ceil(float64(len(recentArr)) / 8))
+		if len(recentArr) > 8 {
+			if len(recentArr) > PageNum*8 {
+				recentArr = recentArr[(PageNum-1)*8 : (PageNum)*8]
+
+			} else {
+				recentArr = recentArr[(PageNum-1)*8 : len(recentArr)]
+
+			}
+		}
+
+		if len(mostViewedArr) > 8 {
+			mostViewedArr = mostViewedArr[0:8]
+		}
+		//for i, a := range recentArr {
+		//	if a.UploadTime == 0 {
+		//		recentArr = append(recentArr[:i], recentArr[i+1:]...)
+		//		fmt.Println(recentArr)
+		//	}
 
 		response := struct {
 			MostViewed   []Video
 			RecentUpload []Video
+			Pages        int
 		}{
-			MostViewed:   mostViewedArr[0:7],
+
+			MostViewed:   mostViewedArr,
 			RecentUpload: recentArr,
+			Pages:        pageAmount,
 		}
 
 		c.JSON(http.StatusOK, gin.H{"message": response})
```

Comment:
Please comment code to allow it to become easier to follow. Overall good logic on this API.